embeddings/data/Dataset.py

```diff
@@ -0,0 +1,9 @@
+from typing import TypeVar, Generic
+from abc import ABC
+
+Data = TypeVar("Data")
+
+
+class DataSet(ABC, Generic[Data]):
```

Comment:
typically class like this is named 'Dataset'

Reply:
hahah I got the same :D